src/symbols/Cluster.tsx

```diff
@@ -238,54 +240,63 @@ export const Cluster: FC<ClusterProps> = ({
           }}
           {...(bind() as any)}
         >
-          <mesh>
-            <ringGeometry attach="geometry" args={[offset, 0, 128]} />
-            <a.meshBasicMaterial
-              attach="material"
-              color={normalizedFill}
-              transparent={true}
-              depthTest={false}
-              opacity={theme.cluster?.fill ? circleOpacity : 0}
-              side={DoubleSide}
-              fog={true}
-            />
-          </mesh>
-          <mesh>
-            <ringGeometry
-              attach="geometry"
-              args={[offset, rad + padding, 128]}
-            />
-            <a.meshBasicMaterial
-              attach="material"
-              color={normalizedStroke}
-              transparent={true}
-              depthTest={false}
-              opacity={circleOpacity}
-              side={DoubleSide}
-              fog={true}
-            />
-          </mesh>
-          {theme.cluster?.label && (
-            <a.group position={labelPosition as any}>
-              {renderLabel ? (
-                renderLabel({
-                  label,
-                  opacity,
-                  fontUrl: labelFontUrl,
-                  theme
-                })
-              ) : (
-                <Label
-                  text={label}
-                  opacity={opacity}
-                  fontUrl={labelFontUrl}
-                  stroke={theme.cluster.label.stroke}
-                  active={false}
-                  color={theme.cluster?.label.color}
-                  fontSize={theme.cluster?.label.fontSize ?? 12}
+          {onRender ? (
+            onRender({
+              label: {
+                position: labelPosition,
+                text: label,
+                opacity: opacity,
+                fontUrl: labelFontUrl
+              },
+              opacity: circleOpacity,
+              offset,
+              rad,
+              padding,
+              theme
+            })
+          ) : (
+            <>
+              <mesh>
+                <ringGeometry attach="geometry" args={[offset, 0, 128]} />
+                <a.meshBasicMaterial
+                  attach="material"
+                  color={normalizedFill}
+                  transparent={true}
+                  depthTest={false}
+                  opacity={theme.cluster?.fill ? circleOpacity : 0}
+                  side={DoubleSide}
+                  fog={true}
                 />
+              </mesh>
+              <mesh>
```

Comment:
Similar to what we did with `Node` and its pre-defined shapes like `Sphere`, if I were you I would look and see if we could abstract these for a user making it easier to implement this w/o knowing Three.